crates/wasmtime/src/config.rs

```diff
@@ -341,6 +346,20 @@ impl Config {
         self
     }
 
+    /// Configures whether or not to reset stack pages after once execution.
+    ///
+    /// Reset stack pages will cost a lot in multi thread environment. It is
+    /// purely a defense-in-depth thing and is not required for correctness.
+    /// In some cases there is no need to care much about safety. You can set
+    /// it to `false`. If you don't know what you are doing, just use the
+    /// default configuration.
```

Comment:
I would suggest this documentation perhaps for this method:

> Configures whether or not stacks used for async futures are reset to zero after usage.
>
> When the `async_support` method is enabled for Wasmtime and the `call_async` variant of calling WebAssembly is used then Wasmtime will create a separate runtime execution stack for each future produced by `call_async`. When using the pooling instance allocator this allocation will happen from a pool of stacks and additionally deallocation will simply release the stack back to the pool. During the deallocation process Wasmtime will by default reset the contents of the stack back to zero.
>
> This reset back to zero, however, is a defense-in-depth mechanism and is not required for correctness. The operation to reset a stack back to zero can be a costly operation in highly concurrent environments. By setting this option to `false` this zero-ing operation will not occur which may prove beneficial to throughput in these environments.

Additionally I might recommend renaming the method/option `async_stack_zeroing` or something like that to make sure the word "async" is somewhere in there along with "zero"